pallets/settlement/src/lib.rs

```diff
@@ -1368,32 +1368,24 @@ impl<T: Trait> Module<T> {
                                 let tx_data = tx_data.remove(2);
                                 let decoded_justified_tx = match tx_data {
                                     MercatTxData::JustifiedTransfer(finalized) => {
-                                        let result = base64::decode(&finalized);
-                                        if result.is_ok() {
-                                            let mut data: &[u8] = &result.unwrap();
-                                            let result = JustifiedTransferTx::decode(&mut data);
-                                            if result.is_ok() {
-                                                result.unwrap()
-                                            } else {
-                                                return Err((
-                                                    leg_id,
-                                                    FailureReason::GetMercatMediatorData,
-                                                ));
-                                            }
-                                        } else {
-                                            return Err((
-                                                leg_id,
-                                                FailureReason::GetMercatMediatorData,
-                                            ));
-                                        }
+                                        base64::decode(&finalized)
+                                            .and_then(|d| {
+                                                Ok(JustifiedTransferTx::decode(&mut &d[..]))
+                                            })
```

Comment:
You can rewrite this as `.map(|d| JustifiedTransferTx::decode(&mut &d[..]))`

------------------

Per monad laws, `monadic_value.flat_map(|x| pure(f(x))` (or `monadic_value >>= pure . f` in Haskell terms) is equivalent to `monadic_value.map(f)`.